### PR TITLE
1D Spectral Element - Finite Volume Remapping

### PR DESCRIPTION
**Pull request by jb-mackay**
Adds finite volume <-> spectral element remapping in 1D.

### CODE REVIEW

File: src/Operators/remapping.jl [hunk 1]
```diff
@@ -57,10 +57,9 @@ function linear_remap_op(target::AbstractSpace, source::AbstractSpace)
...
 
 """
-    overlap_weights(target::T, source::S) where {T <: SpectralElementSpace1D{<:Topology2D, Quadratures.GL{1}},
-            S <: SpectralElementSpace1D{<:Topology2D, Quadratures.GL{1}}}
+    overlap_weights(target, source)
 
-Computes local weights of the overlap mesh for `source_topo` to `target_topo` topologies.
+Computes local weights of the overlap mesh for `source` to `target` spaces.
 
 Returns sparse matrix `W` where entry `W_{ij}` gives the area that target element `i` overlaps source element `j`.
```
Comment (simonbyrne) [inline src/Operators/remapping.jl:64]: Maybe clarify this a bit (e.g. it is really the `i`th global node value)

File: src/Operators/remapping.jl [hunk 2]
```diff
@@ -103,6 +93,66 @@ function overlap_weights(
...
+            ξ, w = Quadratures.quadrature_points(FT, QS_t)
+            x_ov =
+                FT(0.5) * (min_ov + max_ov) .+ FT(0.5) * (max_ov - min_ov) * ξ
+            x_t = FT(0.5) * (min_i + max_i) .+ FT(0.5) * (max_i - min_i) * ξ
+
+            I_mat = Quadratures.interpolation_matrix(x_ov, x_t)
+            for k in 1:Nq_t
+                idx = Nq_t * (i - 1) + k
+                # (integral of basis on overlap) / (reference elem length * overlap elem length)
+                J_ov[idx, j] = w' * I_mat[:, k] ./ 2 * (max_ov - min_ov)
```
Comment (valeriabarra) [inline src/Operators/remapping.jl:138]: Thanks Ben for your valuable contribution!
I would specify in the comment that the k-th column of the interpolant matrix gives you the k-th basis function defined on the element

File: src/Operators/remapping.jl [hunk 3]
```diff
@@ -103,6 +93,66 @@ function overlap_weights(
     return kron(Y_ov, X_ov)
 end
 
+function overlap_weights(
```
Comment (valeriabarra) [inline src/Operators/remapping.jl:96]: Just my little OCD speaking, but I would put the 1D version of this function before the 2D version above. Great work though! Looks great to me

**Comment by valeriabarra** [comment]
Thanks Ben, it looks great to me! 
Did you decide for the 1D spectral field plotting recipe to be in a separate PR, right?

**Review by valeriabarra — APPROVED** [review]

**Comment by jb-mackay** [comment]
> Thanks Ben, it looks great to me! Did you decide for the 1D spectral field plotting recipe to be in a separate PR, right?

Thanks for the feedback, will address the other above comments. Yes, the 1D plotting will be in a later PR.

**Pull request merged**